### DISCUSSION TITLE
Allow "params" for names

### DISCUSSION THREAD

**Discussion by ygoe** [dotnet/csharplang#6816] [Language Ideas]
The `params` keyword is a global keyword so it cannot be used anywhere except for that certain type of parameter definitions. Can we make this a context-sensitive keyword? That would allow having parameters named "params" and other uses of the name. I am aware of the possibility to `@`-prefix a name but would rather avoid that.

**Comment by 333fred** [comment]
We are extremely unlikely to change any global keyword to make it context-sensitive.

**Reply by ygoe** [reply]
Why? Would it break something?

**Reply by CyrusNajmabadi** [reply]
It adds complexity where there is non today. @ is the solution here that we've had functioning since c# 1.0 :-)
